File: Modularity/common/src/main/java/ca/six/common/LoginActivity.java
package ca.six.common;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

public class LoginActivity extends AppCompatActivity implements View.OnClickListener {

    private EditText etName;
    private EditText etPwd;
    private Button btnLogin;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);

        etName = (EditText) findViewById(R.id.etName);
        etPwd = (EditText) findViewById(R.id.etPwd);
        btnLogin = (Button) findViewById(R.id.btnLogin);

        btnLogin.setOnClickListener(this);
    }

    @Override
    public void onClick(View v) {
        String pwd = etPwd.getText().toString();
        String name = etName.getText().toString();
        if("123".equals(pwd) && "szw".equals(name)) {
            Toast.makeText(this, "Login Successfully", Toast.LENGTH_SHORT).show();
            Session.getInstance().user = new User("szw", true);
        } else {
            Toast.makeText(this, "Error name/pwd", Toast.LENGTH_SHORT).show();
        }
    }
}

File: CleanViper/app/src/main/java/ca/six/demo/cleanviper/biz/session/LoginActivity.kt
package ca.six.demo.cleanviper.biz.session

import android.os.Bundle
import android.view.Window
import android.view.WindowManager.LayoutParams.FLAG_FULLSCREEN
import androidx.appcompat.app.AppCompatActivity
import ca.six.demo.cleanviper.R
import ca.six.demo.cleanviper.core.base.BaseActivity
import ca.six.demo.cleanviper.ext.*
import ca.six.demo.cleanviper.router.core.Router
import com.github.ybq.android.spinkit.SpinKitView
import kotlinx.android.synthetic.main.activity_login.*

interface ILoginView {
    fun continueNav()
    fun toast(msg: String)
}

class LoginActivity : BaseActivity(), ILoginView {
    lateinit var presenter: LoginPresenter
    lateinit var spinView: SpinKitView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        requestWindowFeature(Window.FEATURE_NO_TITLE)
        window.setFlags(FLAG_FULLSCREEN, FLAG_FULLSCREEN)
        setContentView(R.layout.activity_login) //上面两个得在加载layout文件之前

        presenter = LoginPresenter(this) //TODO 改为DI
        presenter.disposables = disposables

        btnLogin.setOnClickListener {
            spinView = showProgress()
            presenter.login(etLoginName.stringContent(), etLoginPwd.stringContent())
        }
    }

    override fun continueNav() {
        hideProgress(spinView)
        Router.continueNav(this)
        this.finish()
    }

    override fun toast(msg: String) {
        hideProgress(spinView)
        this.showToast(msg)
    }

}